File: windows/ClaudeTrafficLight/Core/Localization.cs
using System.Globalization;

namespace ClaudeTrafficLight.Core;

/// <summary>
/// Simple, code-embedded localization. Picks strings by system language and
/// falls back to English for unknown languages. Mirrors the macOS L10n tables.
/// </summary>
public sealed record L10n(
    string Working,        // menu row: working
    string Asking,         // menu row: asking a question / waiting
    string Done,           // menu row: done
    string NoSessions,     // header: no active sessions
    string WaitingWord,    // summary counter: "2 <waiting>"
    string WorkingWord,    // summary counter: "1 <working>"
    string DoneWord,       // summary counter: "1 <done>"
    string Refresh,        // menu: refresh
    string Quit,           // menu: quit
    string NotifyTitle,    // notification title: on turning red
    string NotifyMenu,     // menu: notifications (toggle)
    string ActiveSessions, // menu header title (when there are sessions)
    string Hint,           // menu footer hint
    string CheckUpdates,   // menu: manual update check
    string UpToDate,       // update check: already on the latest version
    string UpdateCheckFailed, // update check: network/API problem
    string LocaleId)
{
    /// <summary>Menu label for a given state.</summary>
    public string Label(State state) => state switch
    {
        State.Red => Asking,
        State.Yellow => Working,
        State.Green => Done,
        _ => Done
    };

    /// <summary>Localization for the active UI language (first two letters).</summary>
    public static L10n Current
    {
        get
        {
            string code;
            try { code = CultureInfo.CurrentUICulture.TwoLetterISOLanguageName.ToLowerInvariant(); }
            catch { code = "en"; }
            return Tables.TryGetValue(code, out var l) ? l : English;
        }
    }

    public static readonly L10n English = new(
        "Working…", "Asking a question", "Done",
        "No active Claude sessions",
        "waiting", "working", "done",
        "Refresh", "Quit",
        "Claude is waiting for you", "Notifications",
        "Active sessions", "Click a session to jump to it", "Check for Updates…", "You're up to date",
            "Could not check for updates", "en");

    public static readonly Dictionary<string, L10n> Tables = new()
    {
        ["en"] = English,
        ["tr"] = new(
            "Çalışıyor…", "Soru soruyor", "Bitti",
            "Aktif Claude oturumu yok",
            "bekliyor", "çalışıyor", "bitti",
            "Yenile", "Çıkış",
            "Claude seni bekliyor", "Bildirimler",
            "Aktif oturumlar", "Gitmek için bir oturuma tıkla", "Güncellemeleri Denetle…", "Uygulama güncel",
            "Güncellemeler denetlenemedi", "tr"),
        ["es"] = new(
            "Trabajando…", "Haciendo una pregunta", "Terminado",
            "No hay sesiones de Claude activas",
            "esperando", "trabajando", "terminado",
            "Actualizar", "Salir",
            "Claude te está esperando", "Notificaciones",
            "Sesiones activas", "Haz clic en una sesión para abrirla", "Buscar actualizaciones…", "Estás al día",
            "No se pudo buscar actualizaciones", "es"),
        ["de"] = new(
            "Arbeitet…", "Stellt eine Frage", "Fertig",
            "Keine aktiven Claude-Sitzungen",
            "wartend", "arbeitend", "fertig",
            "Aktualisieren", "Beenden",
            "Claude wartet auf dich", "Benachrichtigungen",
            "Aktive Sitzungen", "Sitzung anklicken, um zu ihr zu springen", "Nach Updates suchen…", "Du bist auf dem neuesten Stand",
            "Updates konnten nicht geprüft werden", "de"),
        ["fr"] = new(
            "En cours…", "Pose une question", "Terminé",
            "Aucune session Claude active",
            "en attente", "en cours", "terminé",
            "Actualiser", "Quitter",
            "Claude vous attend", "Notifications",
            "Sessions actives", "Cliquez sur une session pour y accéder", "Rechercher des mises à jour…", "Vous êtes à jour",
            "Impossible de vérifier les mises à jour", "fr"),
        ["it"] = new(
            "In corso…", "Fa una domanda", "Completato",
            "Nessuna sessione Claude attiva",
            "in attesa", "in corso", "completato",
            "Aggiorna", "Esci",
            "Claude ti sta aspettando", "Notifiche",
            "Sessioni attive", "Clicca una sessione per aprirla", "Controlla aggiornamenti…", "Sei aggiornato",
            "Impossibile controllare gli aggiornamenti", "it"),
        ["pt"] = new(
            "Trabalhando…", "Fazendo uma pergunta", "Concluído",
            "Nenhuma sessão do Claude ativa",
            "aguardando", "trabalhando", "concluído",
            "Atualizar", "Sair",
            "Claude está esperando por você", "Notificações",
            "Sessões ativas", "Clique em uma sessão para abri-la", "Buscar atualizações…", "Você está atualizado",
            "Não foi possível buscar atualizações", "pt"),
        ["ru"] = new(
            "Работает…", "Задаёт вопрос", "Готово",
            "Нет активных сессий Claude",
            "ждут", "работают", "готово",
            "Обновить", "Выход",
            "Claude ждёт вас", "Уведомления",
            "Активные сессии", "Нажмите на сессию, чтобы открыть", "Проверить обновления…", "У вас последняя версия",
            "Не удалось проверить обновления", "ru"),
        ["ja"] = new(
            "作業中…", "質問中", "完了",
            "アクティブな Claude セッションはありません",
            "待機中", "作業中", "完了",
            "更新", "終了",
            "Claude が待っています", "通知",
            "アクティブなセッション", "セッションをクリックして開く", "アップデートを確認…", "最新バージョンです",
            "アップデートを確認できませんでした", "ja"),
        ["zh"] = new(
            "工作中…", "正在提问", "完成",
            "没有活动的 Claude 会话",
            "等待中", "工作中", "完成",
            "刷新", "退出",
            "Claude 正在等你", "通知",
            "活动会话", "点击会话以跳转", "检查更新…", "已是最新版本",
            "无法检查更新", "zh"),
        ["ko"] = new(
            "작업 중…", "질문 중", "완료",
            "활성 Claude 세션 없음",
            "대기 중", "작업 중", "완료",
            "새로고침", "종료",
            "Claude가 기다리고 있습니다", "알림",
            "활성 세션", "세션을 클릭하여 이동", "업데이트 확인…", "최신 버전입니다",
            "업데이트를 확인할 수 없습니다", "ko"),
    };
}

File: macos/Sources/ClaudeStatusCore/Localization.swift
import Foundation

/// Simple, code-embedded localization. Picks strings by system language and
/// falls back to English for unknown languages. (Works with the manual .app
/// packaging — no .lproj / resource bundle required.)
public struct L10n {
    public let working: String     // menu row: working
    public let asking: String      // menu row: asking a question / waiting
    public let done: String        // menu row: done
    public let noSessions: String  // header: no active sessions
    public let waitingWord: String // summary counter: "2 <waiting>"
    public let workingWord: String // summary counter: "1 <working>"
    public let doneWord: String    // summary counter: "1 <done>"
    public let refresh: String     // menu: refresh
    public let quit: String        // menu: quit
    public let notifyTitle: String // notification title: on turning red
    public let notifyMenu: String  // menu: notifications (toggle)
    public let desktopAppMissing: String // feedback: claude:// has no handler (desktop app not installed)
    public let activeSessions: String // menu header title (when there are sessions)
    public let hint: String        // menu footer hint (how to open a session)
    public let checkUpdates: String       // menu: manual update check
    public let upToDate: String           // update check: already on the latest version
    public let updateCheckFailed: String  // update check: network/API problem
    public let localeID: String    // for the relative-time formatter (e.g. "58s ago")

    /// Menu label for a given state.
    public func label(for state: State) -> String {
        switch state {
        case .red:    return asking
        case .yellow: return working
        case .green:  return done
        }
    }

    /// Localization for the active language. macOS 12+ compatible (reads the
    /// language code from `preferredLanguages`).
    public static var current: L10n {
        let pref = Locale.preferredLanguages.first ?? "en"
        let code = String(pref.prefix(2)).lowercased()
        return tables[code] ?? english
    }

    // MARK: - Languages

    public static let english = L10n(
        working: "Working…", asking: "Asking a question", done: "Done",
        noSessions: "No active Claude sessions",
        waitingWord: "waiting", workingWord: "working", doneWord: "done",
        refresh: "Refresh", quit: "Quit",
        notifyTitle: "Claude is waiting for you", notifyMenu: "Notifications",
        desktopAppMissing: "Claude desktop app is not installed — cannot open this session",
        activeSessions: "Active sessions", hint: "Click a session to jump to it",
        checkUpdates: "Check for Updates…", upToDate: "You're up to date",
        updateCheckFailed: "Could not check for updates",
        localeID: "en")

    public static let tables: [String: L10n] = [
        "en": english,
        "tr": L10n(
            working: "Çalışıyor…", asking: "Soru soruyor", done: "Bitti",
            noSessions: "Aktif Claude oturumu yok",
            waitingWord: "bekliyor", workingWord: "çalışıyor", doneWord: "bitti",
            refresh: "Yenile", quit: "Çıkış",
            notifyTitle: "Claude seni bekliyor", notifyMenu: "Bildirimler",
            desktopAppMissing: "Claude masaüstü uygulaması yüklü değil — bu oturum açılamıyor",
            activeSessions: "Aktif oturumlar", hint: "Gitmek için bir oturuma tıkla",
            checkUpdates: "Güncellemeleri Denetle…", upToDate: "Uygulama güncel",
        updateCheckFailed: "Güncellemeler denetlenemedi",
        localeID: "tr"),
        "es": L10n(
            working: "Trabajando…", asking: "Haciendo una pregunta", done: "Terminado",
            noSessions: "No hay sesiones de Claude activas",
            waitingWord: "esperando", workingWord: "trabajando", doneWord: "terminado",
            refresh: "Actualizar", quit: "Salir",
            notifyTitle: "Claude te está esperando", notifyMenu: "Notificaciones",
            desktopAppMissing: "La app de escritorio de Claude no está instalada — no se puede abrir esta sesión",
            activeSessions: "Sesiones activas", hint: "Haz clic en una sesión para abrirla",
            checkUpdates: "Buscar actualizaciones…", upToDate: "Estás al día",
        updateCheckFailed: "No se pudo buscar actualizaciones",
        localeID: "es"),
        "de": L10n(
            working: "Arbeitet…", asking: "Stellt eine Frage", done: "Fertig",
            noSessions: "Keine aktiven Claude-Sitzungen",
            waitingWord: "wartend", workingWord: "arbeitend", doneWord: "fertig",
            refresh: "Aktualisieren", quit: "Beenden",
            notifyTitle: "Claude wartet auf dich", notifyMenu: "Benachrichtigungen",
            desktopAppMissing: "Die Claude-Desktop-App ist nicht installiert — Sitzung kann nicht geöffnet werden",
            activeSessions: "Aktive Sitzungen", hint: "Sitzung anklicken, um zu ihr zu springen",
            checkUpdates: "Nach Updates suchen…", upToDate: "Du bist auf dem neuesten Stand",
        updateCheckFailed: "Updates konnten nicht geprüft werden",
        localeID: "de"),
        "fr": L10n(
            working: "En cours…", asking: "Pose une question", done: "Terminé",
            noSessions: "Aucune session Claude active",
            waitingWord: "en attente", workingWord: "en cours", doneWord: "terminé",
            refresh: "Actualiser", quit: "Quitter",
            notifyTitle: "Claude vous attend", notifyMenu: "Notifications",
            desktopAppMissing: "L'application de bureau Claude n'est pas installée — impossible d'ouvrir cette session",
            activeSessions: "Sessions actives", hint: "Cliquez sur une session pour y accéder",
            checkUpdates: "Rechercher des mises à jour…", upToDate: "Vous êtes à jour",
        updateCheckFailed: "Impossible de vérifier les mises à jour",
        localeID: "fr"),
        "it": L10n(
            working: "In corso…", asking: "Fa una domanda", done: "Completato",
            noSessions: "Nessuna sessione Claude attiva",
            waitingWord: "in attesa", workingWord: "in corso", doneWord: "completato",
            refresh: "Aggiorna", quit: "Esci",
            notifyTitle: "Claude ti sta aspettando", notifyMenu: "Notifiche",
            desktopAppMissing: "L'app desktop di Claude non è installata — impossibile aprire questa sessione",
            activeSessions: "Sessioni attive", hint: "Clicca una sessione per aprirla",
            checkUpdates: "Controlla aggiornamenti…", upToDate: "Sei aggiornato",
        updateCheckFailed: "Impossibile controllare gli aggiornamenti",
        localeID: "it"),
        "pt": L10n(
            working: "Trabalhando…", asking: "Fazendo uma pergunta", done: "Concluído",
            noSessions: "Nenhuma sessão do Claude ativa",
            waitingWord: "aguardando", workingWord: "trabalhando", doneWord: "concluído",
            refresh: "Atualizar", quit: "Sair",
            notifyTitle: "Claude está esperando por você", notifyMenu: "Notificações",
            desktopAppMissing: "O app desktop do Claude não está instalado — não é possível abrir esta sessão",
            activeSessions: "Sessões ativas", hint: "Clique em uma sessão para abri-la",
            checkUpdates: "Buscar atualizações…", upToDate: "Você está atualizado",
        updateCheckFailed: "Não foi possível buscar atualizações",
        localeID: "pt"),
        "ru": L10n(
            working: "Работает…", asking: "Задаёт вопрос", done: "Готово",
            noSessions: "Нет активных сессий Claude",
            waitingWord: "ждут", workingWord: "работают", doneWord: "готово",
            refresh: "Обновить", quit: "Выход",
            notifyTitle: "Claude ждёт вас", notifyMenu: "Уведомления",
            desktopAppMissing: "Приложение Claude для компьютера не установлено — не удаётся открыть эту сессию",
            activeSessions: "Активные сессии", hint: "Нажмите на сессию, чтобы открыть",
            checkUpdates: "Проверить обновления…", upToDate: "У вас последняя версия",
        updateCheckFailed: "Не удалось проверить обновления",
        localeID: "ru"),
        "ja": L10n(
            working: "作業中…", asking: "質問中", done: "完了",
            noSessions: "アクティブな Claude セッションはありません",
            waitingWord: "待機中", workingWord: "作業中", doneWord: "完了",
            refresh: "更新", quit: "終了",
            notifyTitle: "Claude が待っています", notifyMenu: "通知",
            desktopAppMissing: "Claude デスクトップアプリが未インストールのため、このセッションを開けません",
            activeSessions: "アクティブなセッション", hint: "セッションをクリックして開く",
            checkUpdates: "アップデートを確認…", upToDate: "最新バージョンです",
        updateCheckFailed: "アップデートを確認できませんでした",
        localeID: "ja"),
        "zh": L10n(
            working: "工作中…", asking: "正在提问", done: "完成",
            noSessions: "没有活动的 Claude 会话",
            waitingWord: "等待中", workingWord: "工作中", doneWord: "完成",
            refresh: "刷新", quit: "退出",
            notifyTitle: "Claude 正在等你", notifyMenu: "通知",
            desktopAppMissing: "未安装 Claude 桌面应用 — 无法打开此会话",
            activeSessions: "活动会话", hint: "点击会话以跳转",
            checkUpdates: "检查更新…", upToDate: "已是最新版本",
        updateCheckFailed: "无法检查更新",
        localeID: "zh"),
        "ko": L10n(
            working: "작업 중…", asking: "질문 중", done: "완료",
            noSessions: "활성 Claude 세션 없음",
            waitingWord: "대기 중", workingWord: "작업 중", doneWord: "완료",
            refresh: "새로고침", quit: "종료",
            notifyTitle: "Claude가 기다리고 있습니다", notifyMenu: "알림",
            desktopAppMissing: "Claude 데스크톱 앱이 설치되어 있지 않아 이 세션을 열 수 없습니다",
            activeSessions: "활성 세션", hint: "세션을 클릭하여 이동",
            checkUpdates: "업데이트 확인…", upToDate: "최신 버전입니다",
        updateCheckFailed: "업데이트를 확인할 수 없습니다",
        localeID: "ko"),
    ]
}
